interfaces/cython/cantera/func1.pyx

```diff
@@ -87,3 +100,78 @@ cdef class Func1:
 
     def __copy__(self):
         raise NotImplementedError('Func1 object is not copyable')
+
+
+cdef class Tabulated1(Func1):
+    """
+    A `Tabulated1` object representing a tabulated function is defined by sample
+    points and corresponding function values. Inputs are specified either by
+    two iterable objects containing sample point location and function values,
+    or a single array that concatenates those inputs in two columns. Between
+    sample points, values are evaluated based on the keyword argument
+    'interpolation'; options are 'linear' (linear interpolation, default) or
+    'previous' (nearest previous value). Outside the sample interval, the value
+    at the closest end point is returned.
+
+    Examples for `Tabulated1` objects are::
+
+        >>> t1 = Tabulated1([[0, 2], [1, 1], [2, 0]])
+        >>> [t1(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+
+        >>> t2 = Tabulated1([[0, 2], [1, 1], [2, 0]], interpolation='previous')
+        >>> [t2(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 2.0, 1.0, 0.0, 0.0]
+
+        >>> t3 = Tabulated1([0, 1, 2], [2, 1, 0])
+        >>> [t3(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+
+        >>> t4 = Tabulated1(np.array([0, 1, 2]), (2, 1, 0))
+        >>> [t4(v) for v in [-0.5, 0, 0.5, 1.5, 2, 2.5]]
+        [2.0, 2.0, 1.5, 0.5, 0.0, 0.0]
+    """
+
+    def __init__(self, *args, **kwargs):
```

Comment:
There's an implicit requirement here that `interpolation` be a keyword argument, but it's not clear from the docstring that this is a *requirement*. I also wonder whether `method` would be a better name? Especially because this is a keyword argument, `interpolation` is pretty long and probably prone to typos that won't be caught as an argument error (!), leading to really hard to debug behavior.

I think possibly an easier-to-use implementation of this might be:
```python
def __init__(self, arr1=None, arr2=None, *, method='linear'):
```
Then the first two arguments could be provided as positional or keyword, and `method` is checked properly (avoids typos). Probably need to come up with better names than `arr1` and `arr2` though.

Reply:
Not sure about the named `arr1` and `arr2` arguments (leaving `*args`), but :+1: for `method`.